libfreerdp/core/settings.c

```diff
@@ -252,13 +252,27 @@ void settings_load_hkey_local_machine(rdpSettings* settings)
 
 BOOL settings_get_computer_name(rdpSettings* settings)
 {
-	DWORD nSize = MAX_COMPUTERNAME_LENGTH + 1;
-	CHAR computerName[MAX_COMPUTERNAME_LENGTH + 1];
+	DWORD nSize = 0;
+	CHAR* computerName;
+
+	if (GetComputerNameExA(ComputerNameNetBIOS, NULL, &nSize))
```

Comment:
@akallabeth good catch! This is definitely a bug in FreeRDP's GetComputerNameExA implementation. If it uses GetComputerNameA internally (which correctly returns ERROR_BUFFER_OVERFLOW) it must call `SetLastError(ERROR_MORE_DATA)` in that case.